test/ERC1271.t.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {DelegationHandler} from "./utils/DelegationHandler.sol";
+import {IERC1271} from "../src/interfaces/IERC1271.sol";
+
+contract ERC1271Test is DelegationHandler {
+    function setUp() public {
+        setUpDelegation();
+    }
+
+    function test_domainSeparator() public {
+        (
+            ,
+            string memory name,
+            string memory version,
+            uint256 chainId,
+            address verifyingContract,
+            bytes32 salt,
+            uint256[] memory extensions
+        ) = IERC1271(address(signer)).eip712Domain();
+        // Ensure that verifying contract is the signer
+        assertEq(verifyingContract, address(signer));
+        assertEq(abi.encode(extensions), abi.encode(new uint256[](0)));
+        assertEq(salt, bytes32(0));
```

Comment:
assert name and version? or manually insert them in the expected hash below? its just that the hash below is just using already what the 712 domain returns